### PR TITLE
fix: renames in update operations

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
This change fixes all of the Updates that were performing re-names to always setId at the end of the rename operation as well as updating id that was used in the later operations. Also, the `ForceNew`s were removed in cases where rename was also handled in the Update. Test cases were added to show that rename is going through Update and not DeleteAndCreate operation + additional param is always changed to see if other operations are affected by the previous rename operation. The table constraint was adjusted and cannot support the rename for now (until the Read operation is working #2683). Migration guide updated for behavioral changes.

### CODE REVIEW

File: pkg/resources/user_acceptance_test.go [hunk 1]
```diff
@@ -76,6 +76,11 @@ func TestAcc_User(t *testing.T) {
 			// RENAME
 			{
 				Config: uConfig(prefix2, sshkey1, sshkey2),
```
Comment (sfc-gh-asawicki) [inline pkg/resources/user_acceptance_test.go:78]: rename + some other change to validate correctness?

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/resources/function_acceptance_test.go [hunk 2]
```diff
@@ -230,12 +233,54 @@ func TestAcc_Function_migrateFromVersion085(t *testing.T) {
...
+	})
+}
+
+func functionConfig(database string, schema string, name string, comment string) string {
 	return fmt.Sprintf(`
 resource "snowflake_function" "f" {
   database        = "%[1]s"
   schema          = "%[2]s"
   name            = "%[3]s"
+  comment 		  = "%[4]s"
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/function_acceptance_test.go:283]: Hmm, in the IDE it looks the same, but I assume that's spaces vs tabs thing, I'll try to make it equal here.

File: pkg/resources/password_policy_acceptance_test.go [hunk 3]
```diff
@@ -167,3 +180,5 @@ func TestAcc_PasswordPolicyMaxAgeDays(t *testing.T) {
 		},
 	})
 }
+
+// TODO: Tescik (rename + field change)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/password_policy_acceptance_test.go:184]: Removed (the existing test was adjusted)

panic

File: pkg/resources/procedure.go [hunk 4]
```diff
@@ -660,16 +660,17 @@ func UpdateContextProcedure(ctx context.Context, d *schema.ResourceData, meta in
 
 	id := sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(d.Id())
 	if d.HasChange("name") {
-		name := d.Get("name")
-		err := client.Procedures.Alter(ctx, sdk.NewAlterProcedureRequest(id.WithoutArguments(), id.Arguments()).WithRenameTo(sdk.Pointer(sdk.NewSchemaObjectIdentifier(id.DatabaseName(), id.SchemaName(), name.(string)))))
+		newId := sdk.NewSchemaObjectIdentifierWithArguments(id.DatabaseName(), id.SchemaName(), d.Get("name").(string), id.Arguments())
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/procedure.go:663]: Hmm, not sure why it wasn't there, added.

File: pkg/resources/table_acceptance_test.go [hunk 5]
```diff
@@ -1399,9 +1399,12 @@ resource "snowflake_table" "test_table" {
 	return fmt.Sprintf(s, name, databaseName, schemaName)
 }
 
+// TODO: Tescik
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/table_acceptance_test.go:1402]: Existing test adjusted...

delete

File: pkg/resources/user_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-jcieslak) [reply pkg/resources/user_acceptance_test.go:78]: Hmm, here also login_name changes, but I added a comment change for good measure.

File: pkg/resources/file_format_acceptance_test.go [hunk 6]
```diff
@@ -446,6 +448,46 @@ func TestAcc_FileFormat_issue1947(t *testing.T) {
...
+	comment := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	newComment := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	resourceName := "snowflake_file_format.test"
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/file_format_acceptance_test.go:463]: #2716 is not yet merged, I can make the changes after we merge it (or we can address the conflicts there).

[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/file_format_acceptance_test.go:463]: I wrote about this, because this won't be perceived as merge conflict on a git level

**Review by sfc-gh-asawicki — DISMISSED** [review]

File: pkg/resources/function_acceptance_test.go [hunk 7]
```diff
@@ -230,12 +233,54 @@ func TestAcc_Function_migrateFromVersion085(t *testing.T) {
...
+	newComment := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	resourceName := "snowflake_function.f"
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: testAccCheckFunctionDestroy,
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/function_acceptance_test.go]: Added.

**Pull request merged**